src/modules/products/products.controller.ts

```diff
@@ -20,4 +23,16 @@ export class ProductsController {
   async createProduct(@Param('id') id: string, @Body() createProductDto: CreateProductRequestDto) {
     return this.productsService.createProduct(id, createProductDto);
   }
+
+  @UseGuards(OwnershipGuard)
```

Comment:
Based on the implementation of the ownership guard, it requires the organisation id as a parameter. But here it is the product id. This might be an issue, cross reference with the ownership guard implementation please